commcare_connect/multidb/db_router.py

```diff
@@ -0,0 +1,31 @@
+from django.conf import settings
+from django.db import DEFAULT_DB_ALIAS
+
+
+class ConnectDatabaseRouter:
+    """
+    A router to direct migration operations for specific models to the secondary database.
+    """
+
+    def db_for_read(self, model, **hints):
+        # Prevent reads from the secondary database
+        return DEFAULT_DB_ALIAS
+
+    def db_for_write(self, model, **hints):
+        # Prevent writes to the secondary database
+        return DEFAULT_DB_ALIAS
+
+    def allow_relation(self, obj1, obj2, **hints):
+        # Allow relations only in the default database
+        return None
+
+    def allow_migrate(self, db, app_label, model_name=None, **hints):
+        if db == DEFAULT_DB_ALIAS:
+            return True
+        elif db and db == settings.SECONDARY_DB_ALIAS:
```

Comment:
Rather than requiring this for `RunPython`, can we just default it to false in that case? That way no one has to remember to add it and the normal behavior is used. The validation you added is a nice backup option but it would be great if we didn't have to add this everywhere.

Reply:
This has been a tricky thing to get it working. 

The allow_migrate actually has no visibility into what kind of operation (whether RunPython/CreateModel/AddField etc) unless the hints are passed from the operation itself. So we would needed to have an hint called `operation=RunPython`. But instead of passing operation, I have added `run_on_secondary` which can be passed from any migration, though for RunPython, I made it mandatory.